domains/pallets/messenger/src/messages.rs

```diff
@@ -236,7 +232,13 @@ impl<T: Config> Pallet<T> {
             return Err(Error::<T>::InvalidChannelState.into());
         }
 
-        endpoint_handler.message(dst_chain_id, (channel_id, nonce), req)
+        let pre_check_response = if !ChainAllowlist::<T>::get().contains(&dst_chain_id) {
+            Err(Error::<T>::ChainNotAllowed.into())
```

Comment:
While the test detects `reject_transfer/update_transfer_rejected` can be missed due to the `ChainNotAllowed` error, the same can also happen to other errors above e.g. `WeightTagNotMatch`, `MissingChannel`, `InvalidChannelState`, and even the `NoMessageHandler` error from the caller of this function. These should also be handled.

Reply:
Yes that is true and reason its not handled here get the get tests pass and accidentally wont remove chain from allowlist.

Rest of the cases will happen when there are malicious domains which is not the case at the moment for taurus
We should make a proper refactor of this soon but its not a blocker for immediate runtime upgrade